### ISSUE TITLE
Manual removal of Syncoid/Sanoid snapshots is not recursive

### ISSUE THREAD

**Issue by serge-vk** [jimsalterjrs/sanoid#953]
Maybe I miss something obvious, but I have been trying to manually remove some random Syncoid snapshots from my system and I found out accidentally, that I cannot do this recursively: I have `bpool/BOOT@syncoid_...` and all the child snapshots, but when I try removing them by `zfs destroy -r bpool/BOOT@syncoid_...`, only this top snapshot is removed, and I have to manually destroy each of its children. 

The same happens when I try destroying a particular snapshot taken by Sanoid. Only one is actually destroyed even if I specify the `-r` flag.

If I create a test snapshot `zfs snap -r bpool/BOOT@test`, I can remove all the snapshot tree with  `zfs destroy -r bpool/BOOT@test`.

I wonder why this happens.

**Comment by jimsalterjrs** [comment]
If you specify `recursive=zfs` instead of `recursive=yes`, you'll get the behavior you're looking for. 

Sanoid does not typically take snapshots using zfs' built-in recursion, because zfs' built-in recursion can't account for a lot of scenarios revolving around the addition or removal of child snapshots very well.

However, that also means recursive snapshot deletion won't work either. This is why `recursive=zfs` is an option.

**Comment by phreaker0** [comment]
For syncoid snapshots those won't have the same name because of the date/time in the snapshot. Thefore zfs destroy -r won't remove them on childs.

**Comment by serge-vk** [comment]
Thank you for explaining. The interesting thing in this case is that I had `recursive = zfs` set in all my configurations. As it was discovered in the other [issue](https://github.com/jimsalterjrs/sanoid/issues/952#issue-2525989447) that I raised, I had exactly one template that had a duplicate `autosnap = no` setting that upset the configuration. This template was only used for exactly one data set. Nevertheless, _all other data sets_ that did not even use this template were snap-shot as though I had `recursive = yes`. Now I removed all snap-shots and had Sanoid make them again with valid configuration, and they seem to be taken as desired with zfs.
